math/src/main/java/coppercore/math/CachedData.java

```diff
@@ -0,0 +1,77 @@
+package coppercore.math;
+
+public class CachedData<Type> {
+
+    private Type value;
+    private int writeCount;
+    private int maxWrites;
+    private long lastUpdateTime;
+    private long staleTime;
+    private boolean isTimeBased;
+
+    // Constructor for Time-based expiration
+    public CachedData(double staleTime, double change) {
+        this.isTimeBased = true;
+        this.staleTime = (long) (staleTime * 1000); // Convert seconds to milliseconds
+        this.lastUpdateTime = System.currentTimeMillis();
+    }
+
+    // Constructor for WriteCount-based expiration
+    public CachedData(int maxWrites) {
+        this.isTimeBased = false;
+        this.maxWrites = maxWrites;
+        this.writeCount = 0;
+    }
+
+    // Update the value based on the change (for both time and write-based mechanisms)
+    public void update() {
```

Comment:
This should be called from read() or isStale(). We don't want the user to have to manually call update